### DISCUSSION TITLE
Android: Transparent gesture bar

### DISCUSSION THREAD

**Discussion by leo-petrucci** [ionic-team/capacitor#4126] [Q&A]
I've managed to get my notch insets to work, but I'm currently having trouble setting the bottom gesture navigation bar to be transparent so I can have a fully edge-to-edge application.

![Screenshot 2021-01-28 at 16 44 54](https://user-images.githubusercontent.com/23533178/106170216-26d6f400-6188-11eb-9988-cdc7b0db9d08.png)

As you can see the bottom black bar is kind of unsightly.

I'm aware there's no Capacitor API to fix this, but if I wanted to run native Java code to change it when my app starts, where would I put it?

Is `MainActivity.java` the right place? If I extended the class to look like this:

```
import android.os.Bundle;

import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // Add code here
    }
}
```

Would my code run on startup?

Absolutely clueless about native Android code and I'm afraid of breaking Capacitor. Would love to get some advice from someone with more experience with Android.

**Comment by leo-petrucci** [comment]
So, I've been able to:

- Set the UI to fullscreen, which unfortunately completely hides the gesture bar
- Change the color to transparent, which unfortunately renders a yellow color as the app isn't extending below the bar

Still absolutely lost on this.

**Comment by tafelnl** [comment]
You can use this plugin to achieve the behavior you're looking for: https://github.com/capacitor-community/safe-area

**Reply by OSRoot** [reply]
Very nice, there is a slight issue i have encountered in my ionic app, if i enable the safe area it has the status bar hiding  about 10 px of the page (top page)

**Reply by tafelnl** [reply]
If this is still an issue, please open an issue in the repository with some more information and a minimal reproduction
